lib/Parser/LineChecker.php

```diff
@@ -6,21 +6,15 @@
 
 use function in_array;
 use function preg_match;
+use function str_repeat;
 use function strlen;
 use function strpos;
 use function trim;
 
 class LineChecker
 {
     private const HEADER_LETTERS = ['=', '-', '~', '*', '+', '^', '"', '.', '`', "'", '_', '#', ':'];
-
-    /** @var LineDataParser */
-    private $lineParser;
-
-    public function __construct(LineDataParser $lineParser)
-    {
-        $this->lineParser = $lineParser;
-    }
+    public const LIST_MARKER     = '/^([-+*\x{2022}\x{2023}\x{2043}]|(?:[\d#]+\.|[\d#]+\)|\([\d#]+\)))(?: +|$)/u';
```

Comment:
This is btw not a complete set of allowed markers. Enumerated lists can also use `A-Z`, `a-z` and even `I-MMMMCMXCIX`. In order to not make this library overly complex, I only implemented the numeric markers (which was also the only one supported previously by this library).